Story/visit counter for job offers

patriciotella opened the pull request:
US: https://trello.com/c/jQ2Qt6m0/3-como-oferente-quiero-ver-la-cantidad-de-veces-que-mi-oferta-fue-visitada

patriciotella committed:
first commit - visit counter feature created (@wip tagged), visit counter steps with generated ( #pending status)

patriciotella committed:
add first step when i visit de offer twice
add background with preload offer

patriciotella committed:
add second step when i visit de offer page

patriciotella committed:
add last part of de step, when i should see the offer has been visited "x" times

(remove the tag @wip, and i notice that the step passed, so I need to be more precise when I ask for the visit counter)

patriciotella committed:
add visit counter into the model and the view

patriciotella committed:
add visit counter field to job offers table
modify job_offer_spec to check, to validate de new attribute visiti_counter
modify job_offers model, method new_visit (plus one to visit counter)

patriciotella committed:
modify the new_visitor method, now add a visit a persist into the database
add behavior in job_offers controller, now when someone applies to an offer, add one to the visit counter.

patriciotella committed:
add getter for the visit_counter and create the specs to check de behaviour

patriciotella committed:
first scenario complete

patriciotella committed:
second scenario a new offer has no visits

patriciotella committed:
second scenario steps behaviour generated

patriciotella committed:
refactor: delete background from feature, creat new methods in the step definitions to avoid repeating code

romanrizzi commented on app/controllers/job_offers.rb:
```diff
@@ -28,6 +28,7 @@
 
   get :apply, :with =>:offer_id  do
     @job_offer = JobOffer.get(params[:offer_id])
+    @job_offer.new_visitor
```
Mejor usar un nombre como: register_new_visitor

romanrizzi commented on app/views/job_offers/my_offers.erb:
```diff
@@ -5,23 +5,27 @@
 		<th>Location</th>
 		<th>Description</th>
 		<th>Is active</th>
+        <th>Visits</th>
```
Este archivo quedo todo mal indentado

romanrizzi commented on features/step_definitions/visit_counter_for_job_offers_steps.rb:
```diff
@@ -0,0 +1,48 @@
+Given(/^I visit an offer twice$/) do
+  create_new_offer_with 'First Offer','Bernal','The one'
```
Extract method, multiples clicks - codigo repetido

romanrizzi commented on features/step_definitions/visit_counter_for_job_offers_steps.rb:
```diff
@@ -0,0 +1,48 @@
+  click_button('Apply')
+end
+
+When(/^I visit the offer page$/) do
```
Sacar esto y usar el background de que me encuentro logeado

romanrizzi commented on features/step_definitions/visit_counter_for_job_offers_steps.rb:
```diff
@@ -0,0 +1,48 @@
+end
+
+
+def create_new_offer_with a_name,a_location,a_description
```
No le podes pegar a la BD desde cucumber!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

romanrizzi commented on features/step_definitions/visit_counter_for_job_offers_steps.rb:
```diff
@@ -0,0 +1,48 @@
+end
+
+Given(/^I create a new offer$/) do
+  create_new_offer_with 'new job offer','Wilde','Cool job'
```
NO SE PUEDE!!!

romanrizzi commented on features/step_definitions/visit_counter_for_job_offers_steps.rb:
```diff
@@ -0,0 +1,48 @@
+
+def assert_there_is_a_number_of_visits_in_the_correct_row_and_column number_of_visits,a_row, a_column
+  visit '/job_offers/my'
+  within("table tr:nth-child(#{a_row})") do
```
Mal indentado

romanrizzi commented on spec/app/models/job_offer_spec.rb:
```diff
@@ -54,6 +55,19 @@
+  end
+  describe 'visit_counter' do
+
+    let(:job_offer) { JobOffer.new }
```
Codigo repetido

jcalvento commented on app/models/job_offer.rb:
```diff
@@ -46,6 +47,15 @@ def activate
-	end
+  end
+
+  def new_visitor
```
Aclarar de alguna forma que se va a hacer save, o sumarle un visitante y de afuera hacer save

patriciotella committed:
refactor:
1) when the offer register a new visitor do de update for the db from the job_offers_controller, that behaviour did not belong to the model
2)change the way I register a new offer (now I register from the view) and create an extract method to avoid repeating code
3)new method's name from new_visitor to register_new_visitor

patriciotella committed:
refactor #3 removing getter from job_offer model, add default value for visit_counter

patriciotella committed:
removing .to_i from visit_counter , now by default is an Integer

Commit:
Merge branch 'develop' into story/visit_counter_for_job_offers

Conflicts:
	app/models/job_offer.rb
	spec/app/models/job_offer_spec.rb

Commit:
Change migration number and fix cucumber feature
